src/libtls/tests/suites/test_socket.c

```diff
@@ -541,8 +610,26 @@ Suite *socket_suite_create()
 	tcase_add_loop_test(tc, func, 0, \
 						tls_crypto_get_supported_suites(TRUE, version, NULL));
 
+#define add_tls_versions_test(func, from, to) \
+	tcase_add_loop_test(tc, func, from, to+1);
+
 	s = suite_create("socket");
 
+	tc = tcase_create("TLS [1.0..1.3] client to TLS 1.3 server");
+	tcase_add_checked_fixture(tc, setup_creds, teardown_creds);
+	add_tls_versions_test(test_tls_13_server, TLS_1_0, TLS_1_3);
+	suite_add_tcase(s, tc);
+
+	tc = tcase_create("TLS 1.3 client to TLS [1.0..1.2] server");
+	tcase_add_checked_fixture(tc, setup_creds, teardown_creds);
+	add_tls_versions_test(test_tls_13_client, TLS_1_0, TLS_1_3);
```

Comment:
```suggestion
	add_tls_versions_test(test_tls_13_client, TLS_1_0, TLS_1_2);
```
Or change the description so it says `[1.0..1.3] server` (might be kinda redundant, but why not test all combinations).